nerfstudio/fields/kplanes_field.py

```diff
@@ -0,0 +1,409 @@
+# Copyright 2022 The Nerfstudio Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Fields for K-Planes (https://sarafridov.github.io/K-Planes/).
+"""
+
+import itertools
+from dataclasses import field
+from typing import Dict, List, Optional, Tuple, Union
+
+import torch
+import torch.nn.functional as F
+from rich.console import Console
+from torch import nn
+from torchtyping import TensorType
+
+from nerfstudio.cameras.rays import RaySamples
+from nerfstudio.field_components.activations import trunc_exp
+from nerfstudio.field_components.embedding import Embedding
+from nerfstudio.field_components.field_heads import FieldHeadNames
+from nerfstudio.field_components.spatial_distortions import SpatialDistortion
+from nerfstudio.fields.base_field import Field
+from nerfstudio.fields.nerfacto_field import get_normalized_directions
+
+try:
+    import tinycudann as tcnn
+except ImportError:
+    # tinycudann module doesn't exist
+    pass
+
+CONSOLE = Console(width=120)
+
+
+def init_grid_param(grid_nd: int, in_dim: int, out_dim: int, reso: List[int], a: float = 0.1, b: float = 0.5):
```

Comment:
Here `grid_nd` is always 2, it would be cleaner to remove the configuration option and other references.

Reply:
Also `in_dim` can be easily inferred from `reso` (`in_dim = len(reso)`), best to remove it from the arguments.